sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcRelation.scala

```diff
@@ -253,7 +260,7 @@ private[orc] case class OrcTableScan(
     maybeStructOI.map { soi =>
       val (fieldRefs, fieldOrdinals) = nonPartitionKeyAttrs.map {
         case (attr, ordinal) =>
-          soi.getStructFieldRef(attr.name.toLowerCase) -> ordinal
```

Comment:
@chenghao-intel I believe one other problematic place is [this line] [1] in `StandardStructObjectInspector`, which causes ORC to write field names in lowercase to physical ORC files. So we do need to change the object inspector.

[1]: https://github.com/apache/hive/blob/master/serde/src/java/org/apache/hadoop/hive/serde2/objectinspector/StandardStructObjectInspector.java#L58